libr/egg/egg_lang.c

```diff
@@ -559,501 +767,569 @@ eprintf ("CNTXXXPUSH (%s)\n", ctxpush[context-1]);
-		} else elem[elem_n++] = c;
-		elem[elem_n] = '\0';
-	} else {
-		switch (c) {
-		case ';':
-			rcc_next (egg);
-			break;
-		case '"':
-			slurp = '"';
-			break;
-		case '(':
-			slurpin = '(';
-			slurp = ')';
-			break;
-		case '{':
-			if (CTX>0) {
-			//	r_egg_printf (egg, " %s:\n", get_frame_label (0));
-				r_egg_printf (egg, " __begin_%d_%d_%d:\n",
-					nfunctions, CTX, nestedi[CTX]); //%s:\n", get_frame_label (0));
-			}
-			rcc_context (egg, 1);
-			break;
-		case '}':
-			endframe = nested[CTX-1];
-			if (endframe) {
-				// XXX: use endframe[context]
-				r_egg_printf (egg, "%s\n", endframe);
-			//	R_FREE (endframe);
-			}
-			if (CTX>0) {
-				if (nestede[CTX]) {
-					r_egg_printf (egg, "%s:\n", nestede[CTX]);
-					//nestede[CTX] = NULL;
-				} else {
-					r_egg_printf (egg, "  __end_%d_%d_%d:\n",
-						nfunctions, CTX, nestedi[CTX-1]);
-					//get_end_frame_label (egg));
-				}
-				nbrackets++;
-			}
-			rcc_context (egg, -1);
-			if (CTX== 0) {
-				nbrackets = 0;
-				nfunctions++;
-			}
-			break;
-		case ':':
-			if (oc == '\n' || oc == '}')
-				quoteline = '\n';
-			else elem[elem_n++] = c;
-			break;
-		case '#':
-			if (oc == '\n')
-				skipline = 1;
-			break;
-		case '/':
-			if (oc == '/')
-				skipline = 1;
-			else elem[elem_n++] = c;
-			break;
-		default:
-			elem[elem_n++] = c;
-		}
-		if (slurp) {
-			if (elem_n) {
-				ptr = elem;
-				elem[elem_n] = '\0';
-				while (is_space (*ptr)) ptr++;
-				rcc_fun (egg, ptr);
-			}
-			elem_n = 0;
-		}
-	}
-	if (c!='\t' && c!=' ')
-		oc = c;
-	return 0;
+    if (commentmode) {
+        if (c=='/' && oc == '*')
+            commentmode = 0;
+        oc = c;
+        return 0;
+    } else if (c=='*' && oc == '/')
+        commentmode = 1;
+    if (slurp) {
+        if (slurp != '"' && c == slurpin)   // only happend when (...(...)...)
+            exit (eprintf (
+                "%s:%d Nesting of expressions not yet supported\n",
+                file, line));
+        if (c == slurp && oc != '\\') {  // close slurp
+            elem[elem_n] = '\0';
+            if (elem_n > 0)
+                rcc_element (egg, elem);
+            else e->frame (egg, 0);
+            elem_n = 0;
+            slurp = 0;
+        } else elem[elem_n++] = c;
+        elem[elem_n] = '\0';
+    } else {
+        switch (c) {
+        case ';':
+            rcc_next (egg);
+            break;
+        case '"':
+            slurp = '"';
+            break;
+        case '(':
+            slurpin = '(';
+            slurp = ')';
+            break;
+        case '{':
+            if (CTX>0) {
+            //    r_egg_printf (egg, " %s:\n", get_frame_label (0));
+                if (nested_callname[CTX] && strstr(nested_callname[CTX], "if") &&
+                    strstr(elem, "else")) {
+                    *elem = '\x00';
+                    elem_n = 0;
+                    R_FREE (ifelse_table[CTX][nestedi[CTX]-1])
+                    snprintf (str, 64, "  __end_%d_%d_%d:",
```

Comment:
I will rewrite thest "snprintf" tomorrow, sadly, for the comming final exam. Also, add some tests later.